packages/cursorless-vscode/src/ScopeSupportTreeProvider.ts

```diff
@@ -0,0 +1,146 @@
+import { ScopeType } from "@cursorless/common";
+import {
+  ScopeProvider,
+  ScopeSupport,
+  ScopeSupportLevels,
+} from "@cursorless/cursorless-engine";
+import * as vscode from "vscode";
+
+export class ScopeSupportTreeProvider
+  implements vscode.TreeDataProvider<TreeItem>
+{
+  private onDidChangeScopeSupportDisposable: vscode.Disposable | undefined;
+  private treeView: vscode.TreeView<TreeItem>;
+  private supportLevels: ScopeSupportLevels = [];
+
+  private _onDidChangeTreeData: vscode.EventEmitter<
+    TreeItem | undefined | null | void
+  > = new vscode.EventEmitter<TreeItem | undefined | null | void>();
+  readonly onDidChangeTreeData: vscode.Event<
+    TreeItem | undefined | null | void
+  > = this._onDidChangeTreeData.event;
+
+  constructor(
+    private context: vscode.ExtensionContext,
+    private scopeProvider: ScopeProvider,
+  ) {
+    this.treeView = vscode.window.createTreeView("cursorlessScopeSupport", {
+      treeDataProvider: this,
+    });
+
+    this.context.subscriptions.push(
+      this.treeView,
+      this.treeView.onDidChangeVisibility(this.onDidChangeVisible, this),
+      this,
+    );
+  }
+
+  static create(
+    context: vscode.ExtensionContext,
+    scopeProvider: ScopeProvider,
+  ): ScopeSupportTreeProvider {
+    const treeProvider = new ScopeSupportTreeProvider(context, scopeProvider);
+    treeProvider.init();
+    return treeProvider;
+  }
+
+  init() {
+    if (this.treeView.visible) {
+      this.registerScopeSupportListener();
+    }
+  }
+
+  onDidChangeVisible(e: vscode.TreeViewVisibilityChangeEvent) {
+    if (e.visible) {
+      if (this.onDidChangeScopeSupportDisposable != null) {
+        return;
+      }
+
+      this.registerScopeSupportListener();
+    } else {
+      if (this.onDidChangeScopeSupportDisposable == null) {
+        return;
+      }
+
+      this.onDidChangeScopeSupportDisposable.dispose();
+      this.onDidChangeScopeSupportDisposable = undefined;
+    }
+  }
+
+  private registerScopeSupportListener() {
+    this.onDidChangeScopeSupportDisposable =
+      this.scopeProvider.onDidChangeScopeSupport((supportLevels) => {
+        this.supportLevels = supportLevels;
+        this._onDidChangeTreeData.fire();
+      });
+  }
+
+  getTreeItem(element: TreeItem): TreeItem {
```

Comment:
Maybe rename your own type to make it clear that you're not just passing around a vscode tree item